### ISSUE TITLE
Feature request: Bluntness for cutting weapon

### ISSUE THREAD

**Issue by Aerotinge** [CleverRaven/Cataclysm-DDA#49712]
# Is your feature request related to a problem? Please describe.
A forged blade should stay unsharpened, and a used blade may become blunt. On the other hand a whetstone can help a blade render normal, or even more sharpen.

# Describe the solution you'd like
EDIT: rebase on the firearm fouling & mending system, thanks @natsirt721
**Item bluntness**

1. Introducing a "bluntness" level like firearm "fouling" for all cutting weapon
"█ Katana" <- A dull katana, it hardly hurts your finger.
"▄ Katana" <- A half-blunt katana, it's still good for killing zombie but less efficient.
"Katana" <- A newly sharpened katana, it strikes through flesh like a lightning.
The more dull a weapon, the less cutting damage it deals, up to *0.4 if max bluntness reached.
I hope it could be a stepless debuff.

2. Melee and getting dull
A weapon with bluntness will dull after dealing cutting damages. 
The probability of getting dull is determined and modified by its material and material of the enemy.

3. Grinder and getting sharp
I forgot there's already a `grinding` quality, make it more useful.
Add a mending recipe that reduce "bluntness" on an item, requiring a tool with at least grinding 2 and water. 
And time required is based on grinding quality of the tool, hardness of the material, and the length of the blade.
It could be: 
```10 m(base time)*length/30 cm(base length)*material_hardness(1.5 for steel, 2 for hardsteel)```
So a `katana` would cost 3x time and charges comparing to a `knife_combat`
　
Fine sharpening, as "lubricated" of a firearm, can be subdivided later. Whetstones can be ignored for now.

# Describe alternatives you've considered

**Item sharpness**

1. Introducing a code bit thing "Sharpness", for all cutting weapon made with material = `copper`/`brass`/`bronze`/`iron`/`budget_steel`/`steel`/`hardsteel`
sort by hardness, from min. to max. `ceramic` is not available for whetstone process, and the other metals are just too soft.
`superalloy`? Dunno what exactly it is, so it's disqualified here.

2. Sharpness in 5 levels
A newly forged `katana` will be "katana (unsharpened)", while a `knife_combat` dropped by soldier zombie would be like "combat knive (blunt)"

| Sharpness/Buffs    	| Unsharpened 	| Blunt 	| Acceptable 	| Sharp        	| Razor-sharp  	|
|--------------------	|-------------	|-------	|------------	|--------------	|--------------	|
| Cut dmg. buff      	| *.4         	| *.8   	| *1.0       	| *1.1(max +2) 	| *1.2(max +5) 	|
| Buchery skill buff 	| -50         	| -10   	| +0         	| +0           	| +0           	|

3. Melee and getting blunt
A weapon with sharpness will dull after dealing damages. 
The probability of damage is determined and modified by its material.
The blunt-rate could also involve into cut armor of the monster. Soft targets are less likely to dull your blade, but a skeletal juggernaut surely screws it up.
"Razor-sharp" degenerates to "Sharp" quickly, like after 20 hits. 
"Sharp" degenerates to "Acceptable" which takes a bit longer like 50 hits.
"Acceptable" last a long time, like 200-300 hits, then degenerates to "Blunt". And it stops here.

4. Whetstone and getting sharp
Whetstone, imaging this is a `sponge` but only accept for cutting weapon with sharpness.
Instead of soap, it consumes its own charges, and the consumption depends on the length of the target item.
Water is required, but takes only 1 every time.
Sharpening is time consuming, depending on whetstone quality, length of of item and material of the item.
　
Item `whetstone_400`/`polisher` allowing to sharpen an "Unsharpened" weapon to "Blunt" status.
Item `whetstone_1000` allowing to sharpen an "Blunt" weapon to "Acceptable" and then "Sharp" status.
Item `whetstone_3000` allowing to sharpen an `hardsteel`/`steel` made "Acceptable" weapon to "Sharp" and then "Razor-sharp" status.

Whetstones can spawn in house like paper or batteries, but less likely.
Too many whetstone? One `whetstone_kit` that take care of everything is fine.

# Additional context
~~This is somehow a nerf, given that of whetstone are nonrenewable.~~
Some people(me!) heavily rely on Katana and then Nodachi/Naginata in a mid-late game. 
With the bleeding system introducing in 0.F, cutting and piercing weapon become even more meta weapon type.
I think it's time to make some balance on it.

**Comment by robotic-gamer** [comment]
according to Wikipedia apparently flint is a suitable material for making whetstones.

**Comment by ERRORCODE509** [comment]
> given that of whetstone are nonrenewable.

This isn't entirely true. Whetstones are usually made of quartz, but can be made of pretty much any stone, though some are better than others. As such, they'd be pseudo-nonrenewable, as the map generates infinitely, meaning you can mine endless amounts of stone. Seeing how C:DDA is set in New England, granite is plentiful, and is an effective stone for use as a whetstone.
In addition, there are synthetic whetstones, though they would probably be outside the scope of what's possible in-game. However, you don't even need a whetstone to sharpen something. Belt sanders and grinders are used quite commonly nowadays by modern blacksmiths. Theoretically anything abrasive can be used for sharpening, though results would vary.

**Comment by natsirt721** [comment]
This would fit well with the mending system, similar to how firearms gain fouling when not maintained. I'd recommend starting simple for now:

- Three sharpening states (dull, neutral, sharp) instead of five. Three is easier to keep track of. We can assume that the crafting process will produce a blade of at least 'dull' status, which the player can then sharpen further. 
- Introduction of a 'sharpening' quality and tools that possess it. Introduce a few items (innawoods flint-based stone, pre-apoc manual sharpening stone, battery-powered angle grinder/bench grinder) that use it and add the quality to any existing tools that should have it. All tools provide 'Sharpening 1' which serves to let them be used to sharpen blades via the mend system.
- One mend recipe. Restores the blade to 'sharp' status. Requires a tool with Sharpening 1 and possibly oil. Time is based on sharpening quality of the tool, hardness of the material, and the length of the blade. This can be subdivided later, to require minimum sharpening levels for specific materials and perhaps restrict the 'sharp' status to higher level. Survival? skill can also play a role here.

One thing that will need balance is butchering using a blade, how much sharpness is used based on the duration of the activity. I'd leave out general crafting for now, but the 'cut up an item' action could also reduce sharpness.

I could be wrong, but I am under the impression that a dull knife will actually do *more* damage to flesh, as the uneven blade will cause tearing not localized to the point of contact, while a razor-sharp blade will cut a fine-but-deeper gash without damaging much of the surrounding tissue. If this is correct, it would make more sense to me to buff AP, but I don't think that melee weapons can have AP so this is more of a point of consideration than an actual suggestion.

**Comment by Aerotinge** [comment]
Flint, sandpaper or even a sherd of pottery can do roughly sharpen process well. I've put a `polisher` there as a placeholder.

> This isn't entirely true. Whetstones are usually made of quartz, but can be made of pretty much any stone, though some are better than others. As such, they'd be pseudo-nonrenewable, as the map generates infinitely, meaning you can mine endless amounts of stone. Seeing how C:DDA is set in New England, granite is plentiful, and is an effective stone for use as a whetstone.

Yes, pseudo-nonrenewable, that's way more precisely. But I did not think there's still machines and technology to turn raw quartz into a commercial whetstone.

> Theoretically anything abrasive can be used for sharpening, though results would vary.

As I said, a shard of pottery works. However it's incomparable with professional whetstones. If not, why do we have 1000~3000 or even 30000 grit size whetstone IRL?
IRL, I know many chinese katana factory do use industrial grinder to batch sharpening their product before shipment. And it's sharp enough to cut paper or bunch of straw. But some buyers tend to do aftermarket process with a whetstone to make it razor-sharp.

> This would fit well with the mending system, similar to how firearms gain fouling when not maintained. I'd recommend starting simple for now:

Oh why I didn't come up with that? This should fit well.

**Comment by YourCompanionCube** [comment]
> I could be wrong, but I am under the impression that a dull knife will actually do more damage to flesh, as the uneven blade will cause tearing not localized to the point of contact, while a razor-sharp blade will cut a fine-but-deeper gash without damaging much of the surrounding tissue.

This is wrong, especially when the target is wearing clothing. A dull blade doesn't necessarily have a jagged edge, that comes from sharpening without polishing (which gives less edge durability but a "sharper" blade for a bit), a dull blade doesn't cut very well at all since it has rolled the edge over on a microscopic level so it doesn't cut as much as it acts like a fine wedge

[This video](https://youtu.be/uP4wLMmp-8U) shows testing of edge sharpness vs targets with clothing, they are testing some gambeson but even modern clothing like a jacket can put up some resistance to cutting (though not as much as proper armor)

**Comment by natsirt721** [comment]
Right, it makes sense that it is less effective against armor and clothing, I was just referring to actual tissue damage, blade vs bare flesh without anything interposing. My experience is limited to small knives as well, and it definitely makes sense that you want bladed weapons well sharpened.

**Comment by YourCompanionCube** [comment]
as far as i know sharper is also better for tissue damage, though like i said a toothy edge can do what you described but that's not from going dull, it's from sharpening to a coarse grit

**Comment by natsirt721** [comment]
I'm going to take a stab at implementing this. Seems like a pretty straightforward ask, but I'm not very familiar with the code so it may take a bit.

**Comment by actual-nh** [comment]
As I-am-Erk stated in the linked PR, things like this are on hold until there's a system for doing item maintenance that doesn't require player micromanagement. I've added the P5 label so this will stick around until then (not be considered stale and closed).

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/cutting-quality-overhaul/27069/1

**Comment by kevingranade** [comment]
> I've added the P5 label so this will stick around until then

I have the opposite take on this, we know this is desirable and we don't need an issue tracking it until that other system happens.  I promise if this becomes unblocked I will be pushing for this to go in as one of the first set of features related to it.
